### PR TITLE
refactor MergeCoordinator executeBlock to return validation reason

### LINKED ISSUE #3145: refactor MergeCoordinator executeBlock to return validation reason
### Description
[Kintsugi v3 execution spec](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.5/src/engine/specification.md#response) specifies that we should provide additional detail on a block execution that was marked `INVALID`.  Currently `MergeCoordinator.executeBlock()` simply supplies a boolean for success or failure, leveraging `BlockValidator.validateAndProcessBlock`.  

### Acceptance Criteria
* MergeCoordinator.executeBlock returns ValidationResult or similar Optional to indicate the success or failure of the execution, as well as any failure causes
* EngineExecuteBlock incorporates failure causes into the `validationError` field

### PR DESCRIPTION
**Pull request by fab-10**
## Fixed Issue(s)

fixes #3145 
## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java [hunk 1]
```diff
@@ -64,64 +63,70 @@ public MainnetBlockValidator(
...
       badBlockManager.addBadBlock(block);
-      return Optional.empty();
+      return new Result(errMsg);
     }
     final BlockHeader parentHeader = maybeParentHeader.get();
 
     if (!blockHeaderValidator.validateHeader(header, parentHeader, context, headerValidationMode)) {
       badBlockManager.addBadBlock(block);
-      return Optional.empty();
+      return new Result("Invalid header for block " + block.toLogString());
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java:92]: @garyschulte is it enough this generic `Invalid header for block` message or we should popup more details from `blockHeaderValidator.validateHeader`?

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java [hunk 2]
```diff
@@ -64,64 +63,70 @@ public MainnetBlockValidator(
...
+      return new Result(errMsg);
     }
     final MutableWorldState worldState = maybeWorldState.get();
 
-    final Result result = processBlock(context, worldState, block);
+    final BlockProcessor.Result result = processBlock(context, worldState, block);
     if (result.isFailed()) {
       badBlockManager.addBadBlock(block);
-      return Optional.empty();
+      return new Result("Error processing block");
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java:115]: @garyschulte is it enough this generic `Error processing block` message or we should popup more details from `processBlock` method?

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java [hunk 3]
```diff
@@ -64,64 +63,70 @@ public MainnetBlockValidator(
...
     }
 
     List<TransactionReceipt> receipts = result.getReceipts();
     if (!blockBodyValidator.validateBody(
         context, block, receipts, worldState.rootHash(), ommerValidationMode)) {
       badBlockManager.addBadBlock(block);
-      return Optional.empty();
+      return new Result("Block body not valid");
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java:122]: @garyschulte same here, is it enough this generic `Block body not valid` message or we should popup more details from `blockBodyValidator.validateBody` method?

Comment (garyschulte) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java:122]: "Block body not valid" is good enough for the error message we are returning to the consensus client.  More detail is better but it seems wise to limit the scope of the signature changes.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java (hunk 2 above)
Comment (garyschulte) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java:115]: good enough IMO.  If we need more specifics we can always implement it, but the api we are trying to support is not user facing nor is the consumer of this api user facing.  The only real motivation is better logging IMO

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java (hunk 1 above)
Comment (garyschulte) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java:92]: :+1:

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java (hunk 3 above)
Comment (fab-10) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java]: same as above

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java (hunk 2 above)
Comment (fab-10) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java]: same as above

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java (hunk 1 above)
Comment (fab-10) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java]: Ok, then we can stop here for the moment, I have just made un update to extract the failure handling code in a method, so the code is cleaner.

**Review by garyschulte — COMMENTED** [review]
posthumous 👍

**Pull request merged**